src/ctap/data_formats.rs

```diff
@@ -676,22 +670,76 @@ impl TryFrom<CoseKey> for ecdh::PubKey {
         if algorithm != ECDH_ALGORITHM && algorithm != ES256_ALGORITHM {
             return Err(Ctap2StatusCode::CTAP2_ERR_UNSUPPORTED_ALGORITHM);
         }
-        let curve = extract_integer(ok_or_missing(curve)?)?;
-        if curve != P_256_CURVE {
-            return Err(Ctap2StatusCode::CTAP2_ERR_UNSUPPORTED_ALGORITHM);
+
+        Ok(CoseKey {
+            x_bytes: *array_ref![x_bytes.as_slice(), 0, ecdh::NBYTES],
+            y_bytes: *array_ref![y_bytes.as_slice(), 0, ecdh::NBYTES],
+            algorithm,
+        })
+    }
+}
+
+impl From<CoseKey> for cbor::Value {
+    fn from(cose_key: CoseKey) -> Self {
+        let CoseKey {
+            x_bytes,
+            y_bytes,
+            algorithm,
+        } = cose_key;
+
+        cbor_map! {
+            1 => EC2_KEY_TYPE,
+            3 => algorithm,
+            -1 => P_256_CURVE,
+            -2 => x_bytes,
+            -3 => y_bytes,
         }
-        let x_bytes = extract_byte_string(ok_or_missing(x_bytes)?)?;
-        if x_bytes.len() != ecdh::NBYTES {
-            return Err(Ctap2StatusCode::CTAP1_ERR_INVALID_PARAMETER);
+    }
+}
+
+impl From<ecdh::PubKey> for CoseKey {
+    fn from(pk: ecdh::PubKey) -> Self {
+        let mut x_bytes = [0; ecdh::NBYTES];
+        let mut y_bytes = [0; ecdh::NBYTES];
+        pk.to_coordinates(&mut x_bytes, &mut y_bytes);
+        CoseKey {
+            x_bytes,
+            y_bytes,
+            algorithm: ECDH_ALGORITHM,
         }
-        let y_bytes = extract_byte_string(ok_or_missing(y_bytes)?)?;
-        if y_bytes.len() != ecdh::NBYTES {
-            return Err(Ctap2StatusCode::CTAP1_ERR_INVALID_PARAMETER);
+    }
+}
+
+impl From<ecdsa::PubKey> for CoseKey {
+    fn from(pk: ecdsa::PubKey) -> Self {
+        let mut x_bytes = [0; ecdh::NBYTES];
+        let mut y_bytes = [0; ecdh::NBYTES];
+        pk.to_coordinates(&mut x_bytes, &mut y_bytes);
+        CoseKey {
+            x_bytes,
+            y_bytes,
+            algorithm: ES256_ALGORITHM,
         }
+    }
+}
+
+impl TryFrom<CoseKey> for ecdh::PubKey {
+    type Error = Ctap2StatusCode;
 
-        let x_array_ref = array_ref![x_bytes.as_slice(), 0, ecdh::NBYTES];
-        let y_array_ref = array_ref![y_bytes.as_slice(), 0, ecdh::NBYTES];
-        ecdh::PubKey::from_coordinates(x_array_ref, y_array_ref)
+    fn try_from(cose_key: CoseKey) -> Result<Self, Ctap2StatusCode> {
+        let CoseKey {
+            x_bytes,
+            y_bytes,
+            algorithm,
+        } = cose_key;
+
+        // Since algorithm can be used for different COSE key types, we check
+        // whether the current type is correct for ECDH. For an OpenSSH bugfix,
```

Comment:
Can you document which OpenSSH bug you're referring to?

Reply:
I mentioned #90 in the comment.

Reply:
It might be preferable to expand it to the full URL for clarity: `https://github.com/google/OpenSK/issues/90`